pages/learn/develop/container-contracts.md

```diff
@@ -0,0 +1,112 @@
+---
+title: Container contracts
+excerpt: A guide to using container contracts to enforce device compatibility
+---
+
+# Container contracts
+
+**Note** Container contracts are available for devices running Supervisor versions >= 10.6.17. All prior versions will not enforce any contracts.
+
+Container contracts are used to ensure the compatibility of a device to run an individual service of an application. For example, container contracts may be used to ensure that the device is running a specific version of [{{ $names.company.lower }} Supervisor][supervisor] or has a specific version of the [NVIDIA Tegra Linux Driver Package][l4t] (L4T).
+
+Each service container can define the contracts that it requires. If a container's contract requirements are not met, the release is not deployed to the device, unless it is marked as [optional](#optional-containers). You only need to define a contract for service containers that you wish to enforce a contract upon. Currently, you may define contract requirements based on the versions of Supervisor and L4T.
```

Comment:
Is "service container" a common phrase in the Balena world?

"Currently, you may define contract requirements based on the versions of Supervisor and L4T."

Presumably that list will be expanded in future? Is it worth mentioning that?

Reply:
As noted below I've made this more consistent using services. I think anything planned for the future can be left out of docs for now and would likely be covered in a blog or similar.